### PR TITLE
Unit tests for cms.navigation

### PR DESCRIPTION
**Pull request by vxsx**
22% → 100%

reopen from https://github.com/FinalAngel/django-cms/pull/177

additionally changes the location for downloading phantomjs (reliable one)
and splits up integration tests into 3 buckets so it's faster and less error prone

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -175,36 +175,57 @@ gulp.task('tests:unit:watch', function () {
...
+            'dragndrop',
+            'history',
+            'revertLive',
+            'narrowScreen'
+        ]
     ];
+
+    var files = [];
+
+    switch (process.env.INTEGRATION_TESTS_BUCKET) {
```
Comment (FinalAngel) [inline gulpfile.js]: this probably might need some comments

File: gulpfile.js [hunk 2]
```diff
@@ -175,36 +175,57 @@ gulp.task('tests:unit:watch', function () {
...
-        'disableToolbar',
-        'dragndrop',
-        'history',
-        'revertLive',
-        'narrowScreen',
-        'clipboard',
-        'modal',
-        'pagetree'
+    var buckets = [
+        [
```
Comment (FinalAngel) [inline gulpfile.js]: dunno here what bucket is used for what case

File: cms/tests/frontend/unit/fixtures/toolbar.html [hunk 3]
```diff
@@ -5,6 +5,61 @@
                 <a href="/" title="cms_version"><span>django CMS</span></a>
             </div>
             <ul class="cms-toolbar-item cms-toolbar-item-navigation">
+                <li>
+                    <a href="">
+                        <span>example.com<span class="cms-icon cms-icon-arrow"></span></span>
+                    </a>
+                    <ul>
+                        <li class=" ">
```
Comment (FinalAngel) [inline cms/tests/frontend/unit/fixtures/toolbar.html]: I assume the space in class is cause of copy & paste?

**Comment by FinalAngel** [comment]
:+1: nice work

File: cms/tests/frontend/unit/fixtures/toolbar.html (hunk 3 above)
Comment (vxsx) [inline cms/tests/frontend/unit/fixtures/toolbar.html]: yup. didn't think it would make sense to prettify fixture

File: gulpfile.js (hunk 2 above)
Comment (vxsx) [inline gulpfile.js]: what do you mean? just three buckets, ordered list of buckets

File: gulpfile.js (hunk 1 above)
Comment (vxsx) [inline gulpfile.js]: added comment, is it more clear that way?

**Pull request merged**